lib/build.ts

```diff
@@ -281,32 +286,18 @@ async function checkAllowDockerPlatformHandling(
 		);
 		if (fromInstructions.length === 0) {
 			// Sanity check
-			console.log(
-				'Build task associated Dockerfile does no reference any images',
-			);
+			debug(`${task.serviceName}: Dockerfile does not reference any images`);
 			return false;
 		}
-		imageReferences = [];
-		fromInstructions.forEach((inst) => {
-			const fromLineArguments = inst.getArguments();
-			if (fromLineArguments.length === 0) {
-				// bail out, no image specified?
-				return;
-			}
-
-			let imgRef = fromLineArguments[0].getValue();
-
-			// hacky way to ignore the --platform flag in the FROM line
-			// In that case, just get the next argument :-p
-			if (imgRef.startsWith('--platform')) {
-				if (fromLineArguments.length < 2) {
-					// bail out, no image specified?
-					return;
+		for (const inst of fromInstructions) {
+			for (const arg of inst.getArguments()) {
+				const val = arg.getValue();
+				if (!val.startsWith('--')) {
+					imageReferences.push(val);
+					break;
```

Comment:
There was no bug here, but the benefits of this change are:
* Future proof against other FROM `--` options that may be added in the future, besides `--platform`.
* Fewer lines of code and, in my biased opinion, easier to read.